src/main/java/org/openstreetmap/atlas/checks/validation/areas/PedestrianAreaOverlappingEdgeCheck.java

```diff
@@ -0,0 +1,245 @@
+                        || (intersectsAtEnd && !intersectsAtStart
+                                && DISTANCE_GREATER_THAN_ONE_METER.test(startLocDistance)
+                                && areaPolygon
+                                        .fullyGeometricallyEncloses(edgeSegmentStartLocation)))
+                {
+                    // Collect all connected edges that are within the pedestrian area
+                    edge.connectedEdges().stream().filter(connectedEdge -> this
+                            .isValidIntersectingEdge(connectedEdge, area)
+                            && areaPolygon.fullyGeometricallyEncloses(connectedEdge.asPolyLine()))
+                            .forEach(overlappingEdges::add);
+                    // Add the intersecting edge as well to the set
+                    overlappingEdges.add(edge);
+                }
+            });
+        }
+        if (!overlappingEdges.isEmpty())
+        {
+            this.markAsFlagged(object.getOsmIdentifier());
+            final CheckFlag flag = this.createFlag(overlappingEdges,
+                    this.getLocalizedInstruction(0, object.getOsmIdentifier(),
+                            new StringList(getIdentifiers(overlappingEdges)).join(", ")));
+            flag.addObject(object);
+            return Optional.of(flag);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Collects all atlas identifiers of given set of {@link AtlasObject}s
+     *
+     * @param objects
+     *            set of {@link AtlasObject}s
+     * @return {@link Iterable<String>} containing the atlas identifiers of input objects
+     */
+    private Iterable<String> getIdentifiers(final Set<AtlasObject> objects)
+    {
+        return Iterables.stream(objects).map(AtlasObject::getIdentifier).map(String::valueOf)
+                .collectToList();
+    }
+
+    /**
+     * Checks if the given segment overlaps any of the segments in the given set
+     *
+     * @param segments
+     *            set of segments
+     * @param segment
+     *            any given segment
+     * @return true if the given segment overlaps any of the segments in the set
+     */
+    private boolean isNotSnappedToSegments(final List<Segment> segments, final Segment segment)
+    {
+        return segments.stream().noneMatch(areaSegment -> areaSegment.overlapsShapeOf(segment));
+    }
+
+    /**
+     * Checks if the given edge and area have the same elevation or not.
+     *
+     * @param edge
+     *            any given edge
+     * @param area
+     *            any given area
+     * @return true if the edge and area have the same Layer and Location tag values
+     */
+    private boolean isOfSameElevation(final Edge edge, final Area area)
+    {
+        return area.getTag(LayerTag.KEY).orElse("").equals(edge.getTag(LayerTag.KEY).orElse(""))
```

Comment:
I'm imagining there might be some issues when (say) the Edge is marked at Level 1, and the Area doesn't have a Level tag, but "should" be Level 1. It sounds like the FP for this check was low so I think this might be an uncommon scenario but maybe something to keep in mind for a future enhancement/exploration

Reply:
If the edge is marked as level 1 and area has no level tag, then the comparison is between "1" and empty string. So I think we should be fine.

Reply:
Right that would be the comparison, but in that case it would return ```false``` for ```isOfSameElevation``` when we might want ```true```. We'd want ```true``` in the event that an area/edge is improperly marked as not having a Level tag, but actually is at Level 1 (confirmable in the editor). I guess it boils down to how we want to handle comparisons between a tagged area/edge and an untagged area/edge.

Reply:
I see what you are saying. I think it is a valid suggestion. I will confirm with Mara and go from there.

Reply:
Just for clarification, a feature without a layer tag is assumed to be at layer 0, not 1.

Reply:
https://github.com/sayas01/atlas-checks/pull/13/commits/684f1c05484006fae450b6ca22e1c57ea86ade38 has the updates based on the comment. LocationTag has been removed from the requirement because of its ambiguity. For this check, same elevation is now determined only using level and layer tags.